packages/mst-middlewares/src/action-logger.ts

```diff
@@ -1,12 +1,14 @@
-import { getPath, IMiddlewareEvent } from "mobx-state-tree"
+import * as mst from "mobx-state-tree"
```

Comment:
I think `* as ` imports make tree shaking worse?

Reply:
https://developers.google.com/web/fundamentals/performance/optimizing-javascript/tree-shaking/

Reply:
just wanted to avoid the possibility of missing an import of an internal type when using TS <= 2.9 (although in theory that's not supported anymore), but guess I was just lazy :) will change

Reply:
ok, just checked it. it is kind of annoying, since for TS to pick up types that are being used internally you have to actually import them from the internal path

e.g. atomic uses (indirectly) IMiddlewareHandler, so I tried:
`
// not actually used but required for proper typing generation
import { IMiddlewareHandler } from "mobx-state-tree"
`
which still generates
`
declare const atomic: import("mobx-state-tree/dist/core/action").IMiddlewareHandler;
export default atomic;
`

so to fix it I have to mark the inferred types as actually `IMiddlewareHandler`, which for action middleware might be trivial, but for the redux one it is not.

by using the star import this is not needed and less error prone (and anyway in the old middlewares it was already importing all of mst since the redux middleware actually uses the star import.

so the question is, should I make the code uglier by adding all these manual type annotations to allow tree shaking or is the star import ok?

Reply:
Hmm, I take that back, both using star import and not using it end up in the same "relative" import.

I will do the tree shaking, though the question now is, should I:
- painfully import and mark inferred types as imported to support TS <= 2.8 (which in theory is not supported anymore)
- let TS generate the relative imports